nir/ir.py

```diff
@@ -166,6 +166,19 @@ def __post_init__(self):
         self.output_type = {"output": np.array(self.weight.shape)[[0, 2, 3]]}
 
 
+@dataclass
+class SumPool2d(NIRNode):
+    """Sum pooling layer in 2d."""
+
+    kernel_size: np.ndarray  # (Height, Width)
+    stride: np.ndarray  # (Height, width)
+    padding: np.ndarray  # (Height, width)
+
+    def __post_init__(self):
+        self.input_type = {"input": ()}
```

Comment:
Could we do something similar to https://github.com/neuromorphs/NIR/blob/main/nir/ir.py#L143 ?

Reply:
The shape of the input cannot be inferred from the parameters of pooling or convolutional layers .. at least in the case of conv2d, we can specify the channels of the input and output. Height and width are arbitrary. (it is incorrectly defined in the post init code of conv2d btw.) In case of pooling, there is no constraint on any of the dimensions. The most you can say is that the input needs to be 2/3 dimensional.

This is also why I am not warn to the notion of adding these things in the node description.

Reply:
I see your point about the shapes. My thinking comes from the "disambiguate as much as possible"-principle. I agree with what you're saying, but I also don't have any good ways to resolve this, I find. This doesn't break anything, so maybe we should just get going? And then think about how to deal with shapes going forward.